.circleci/config.yml

```diff
@@ -528,13 +535,45 @@ parameters:
   study_guid:
     type: string
     default: "UNKNOWN"
+  api_call:
+    type: enum
+    enum: ["build-and-store","deploy","UNKNOWN"]
+    default: "UNKNOWN"
+  deploy_env:
+    type: enum
+    enum: ["dev", "test", "staging", "prod", "UNKNOWN"]
+    default: "UNKNOWN"
   do-builds:
     type: boolean
     default: false
 
 workflows:
   version: 2
 
+  api-build-and-store-workflow:
+    description: "Build study specified in api call"
+    when:
+      and:
+        - equal: ["build-and-store", << pipeline.parameters.api_call >>]
+        - not:
+            equal: ["UNKNOWN", << pipeline.parameters.study_key >>]
+    jobs:
+      - app-build-and-store-job
+
+  api-deploy-workflow:
+    description: "Deploy a study build to environment specified in api call"
+    when:
+      and:
+        - equal: ["deploy", << pipeline.parameters.api_call >>]
+        - not:
+            equal: ["UNKNOWN", << pipeline.parameters.deploy_env >>]
+        - << pipeline.parameters.study_key >>
+        - not:
+            equal: ["UNKNOWN", << pipeline.parameters.study_key >>]
+    jobs:
+      - deploy-stored-build-job:
+          study_key: << pipeline.parameters.study_key >>
+
```

Comment:
The UNKNOWN values are used to figure out whether this is a workflow being initiated by a script or by a github commit event.